BookingProject/BookingWebAPI/Payment.Infrastructure/Repositories/UserBalanceRepository.cs

```diff
@@ -0,0 +1,99 @@
+﻿using Contracts.Clients;
+using Microsoft.EntityFrameworkCore;
+using Microsoft.Extensions.Logging;
+using Payment.Domain.Models;
+using Payment.Infrastructure.DataContext;
+using Payment.Infrastructure.Interfaces;
+using Redis;
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+
+namespace Payment.Infrastructure.Repositories
+{
+    public class UserBalanceRepository(
+            PaymentDbContext _context,
+            ILogger<UserBalanceRepository> _logger,
+            IMonolithClient _monolithClient,
+            IRedisCacheService _cacheService
+        ) : IUserBalanceRepository
+    {
+        public async Task<bool> ChangeBalanceAsync(string adminId, string userId, string currencyName, decimal newBalance)
+        {
+            var role = await  _monolithClient.GetRoleByIdAsync(adminId);
```

Comment:
U can just add the same JWT auth as present in the project.Also u can resolve auth on reverse proxy level.But no need to get roles from monolith, u already have them in token